modules/tensor_mechanics/doc/content/syntax/NEML2/index.md

```diff
@@ -0,0 +1,98 @@
+# NEML2 syntax
+
+The `NEML2` block is used to construct a set of material and userobjects so that MOOSE can "outsource" the material update to NEML2. Different objects get created depending on the operation mode (explained below). They are summarized in the following table for convenience:
+
+| Object                        | Type       | Operation mode | Description                                                                                  |
+| ----------------------------- | ---------- | -------------- | :------------------------------------------------------------------------------------------- |
+| CauchyStressFromNEML2         | Material   | ELEMENT        | Perform element-wise batched material update                                                 |
+| CauchyStressFromNEML2UO       | UserObject | ALL            | Perform mesh-wise batched material update                                                    |
+| CauchyStressFromNEML2Receiver | Material   | ALL            | Assign the outputs of the batched material update into the corresponding material properties |
+
+## Example Input Syntax
+
+!listing test/tests/neml2/fem.i block=NEML2
+
+The field `input` specifies the relative path to the NEML2 input file. The field `model` tells MOOSE which material model to import from the NEML2 input file. The field `device` specifies where to evaluate the NEML2 model, e.g., CPU or CUDA. The parameter `mode` determines the mode of operation for NEML2, and it is important to understand the differences between the modes in order to use NEML2 most efficiently. Each mode is discussed below in detail.
```

Comment:
use `[!param]()` syntax, picking one of the 3 objects

Reply:
why not pick the action?

Reply:
did not think about it ;D